libs/shared/modal/src/lib/shared-modal.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react';
+import Modal from '@material-ui/core/Modal';
+import Backdrop from '@material-ui/core/Backdrop';
+import Fade from '@material-ui/core/Fade';
+import Button from '@material-ui/core/Button';
+
+import './shared-modal.scss';
+
+/* eslint-disable-next-line */
+export interface SharedModalProps {}
+
+export const SharedModal = (props: SharedModalProps) => {
+  const [open, setOpen] = useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
```

Comment:
Instead of handleOpen and handleClose you could create one method:  handleToggle( value ) {}